### ISSUE TITLE
🚀 [Feature]: Implement Valkey Storage

### ISSUE THREAD

**Issue by H0llyW00dzZ** [gofiber/storage#1361]
### Feature Description

This is similar to the Redis link [`here`](https://github.com/valkey-io/valkey-go)

### Checklist:

- [X] I agree to follow Fiber's [Code of Conduct](https://github.com/gofiber/fiber/blob/master/.github/CODE_OF_CONDUCT.md).
- [X] I have checked for existing issues that describe my suggestion prior to opening this one.
- [X] I understand that improperly formatted feature requests may be closed without explanation.

**Comment by gaby** [comment]
@H0llyW00dzZ The Redis and Rueidis clients work with Valkey.
- https://github.com/gofiber/storage/blob/main/redis/README.md
- https://github.com/gofiber/storage/blob/main/rueidis/README.md

**Comment by H0llyW00dzZ** [comment]
> @H0llyW00dzZ The Redis and Rueidis clients work with Valkey. Therr's no need for anlther driver.
> 
> * https://github.com/gofiber/storage/blob/main/redis/README.md
> * https://github.com/gofiber/storage/blob/main/rueidis/README.md
> 
> The valkey-go is a fork of rueidis which we have a storage driver for.

However, that might not work anymore in the future.

**Comment by grivera64** [comment]
I think we are at a good point to revisit this issue. According to a [December 2024 Redis Blog Post](https://redis.io/blog/what-is-valkey/):

> And though [Redis and Valkey] were both once based on the same code, they’ve begun to diverge in feature sets,
> functionality, capabilities, contributors, and more. Over time, the divergence is only likely to increase. 

In this case, it may make sense to officially support Valkey through a separate storage driver. If we do this, we don't need to worry as much about breaking changes between Redis/Rueidis clients and Valkey clients for users that rely on GoFiber's storage module.

Currently, the Valkey client appears to be just enough of the same as Ruedis, so it could be as simple as a copy-and-paste of the existing driver and rename to use Valkey's syntax. This way, once the clients start to vastly diverge, it won't inadvertently break GoFiber applications that upgrade to a future version.

What do you all think of this?

**Comment by gaby** [comment]
@grivera64 Agree, the code from the `rueidis` driver can be copy/pasted, and swap the dependency for https://github.com/valkey-io/valkey-go

**Comment by H0llyW00dzZ** [comment]
nice, finally

**Comment by gaby** [comment]
@ReneWerner87 We need a release for Valkey

**Comment by H0llyW00dzZ** [comment]
> @ReneWerner87 We need a release for Valkey

Furthermore, it can improve performance related to Valkey, such as reducing latency caused by regional factors. In contrast to Redis Cloud, where database deployment for caching or as an in-memory database is restricted to specific regions, Valkey can be self-hosted or directly accessed from cloud providers like Google Cloud.

**Comment by gaby** [comment]
@H0llyW00dzZ You can self-host Redis too.

I run multiple self-hosted Redis and Redis Cluster instances.

**Comment by ReneWerner87** [comment]
> @ReneWerner87 We need a release for Valkey

is a 0.0.1 version okay, as there will definitely be a breaking change in the next month due to our adaptation to fiber v3 ?

**Comment by H0llyW00dzZ** [comment]
> @H0llyW00dzZ You can self-host Redis too.
> 
> I run multiple self-hosted Redis and Redis Cluster instances.

@gaby Do you self-host Redis with TLS or not? Beause Self-hosting Redis has limitations compared to using Redis Cloud.

**Comment by gaby** [comment]
> > @ReneWerner87 We need a release for Valkey
> 
> is a 0.0.1 version okay, as there will definitely be a breaking change in the next month due to our adaptation to fiber v3 ?

Yes

**Comment by ReneWerner87** [comment]
https://github.com/gofiber/storage/releases/tag/valkey%2Fv0.1.0

**Comment by gaby** [comment]
> > @H0llyW00dzZ You can self-host Redis too.
> > I run multiple self-hosted Redis and Redis Cluster instances.
> 
> @gaby Do you self-host Redis with TLS or not? Beause Self-hosting Redis has limitations compared to using Redis Cloud.

With and without TLS. It's the same thing. If you want all the features you have to use Redis Stack. See here https://github.com/redis-stack/redis-stack

**Comment by H0llyW00dzZ** [comment]
> > @ReneWerner87 We need a release for Valkey
> 
> is a 0.0.1 version okay, as there will definitely be a breaking change in the next month due to our adaptation to fiber v3 ?

@ReneWerner87 It will be better for both v2 and v3 because Fiber v2 is stable. I've been using it on Kubernetes.
